examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1570,16 +1655,23 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
                     )
                     prompt_embeds_input = prompt_embeds.repeat(elems_to_repeat_text_embeds, 1, 1)
                     model_pred = unet(
-                        noisy_model_input,
+                        inp_noisy_latents if args.do_edm_style_training else noisy_model_input,
                         timesteps,
                         prompt_embeds_input,
                         added_cond_kwargs=unet_added_conditions,
                         return_dict=False,
                     )[0]
 
+                if args.do_edm_style_training:
+                    model_pred = noise_scheduler.precondition_outputs(noisy_model_input, model_pred, sigmas)
+                    # We are not doing weighting here because it tends result in numerical problems.
+                    # See: https://github.com/huggingface/diffusers/pull/7126#issuecomment-1968523051
+                    # There might be other alternatives for weighting as well:
+                    # https://github.com/huggingface/diffusers/pull/7126#discussion_r1505404686
+
                 # Get the target for loss depending on the prediction type
                 if noise_scheduler.config.prediction_type == "epsilon":
-                    target = noise
+                    target = model_input if args.do_edm_style_training else noise
```

Comment:
We should do this for `v-predition` as well, in EDM, the target is always the actual input as always denoise the image in `precondition_outputs`

Reply:
That gives the velocity, but when using EDM, whether we are predicting `epsilon` or `v`, it is always converted to back to the sample, hence the target should be `model_input`. Loss is always computed against the actual image and predicted image and not `v` or `eps` . Does that make sense ?

Reply:
Makes sense, thanks! However, EDMEuler doesn't have a `get_velocity()` function. Should we maybe tackle this is a separate PR and throw an error here instead?

Reply:
Done in dde75950.